packages/components/psammead-navigation/src/Dropdown/index.jsx

```diff
@@ -0,0 +1,211 @@
+  script,
+  currentPageText,
+  active,
+  service,
+  url,
+  dir,
+}) => (
+  <DropdownLi role="listitem">
+    {active && currentPageText ? (
+      <StyledDropdownLink script={script} service={service} href={url}>
+        <StyledCurrentLink
+          dir={dir}
+          // eslint-disable-next-line jsx-a11y/aria-role
+          role="text"
+        >
+          <VisuallyHiddenText>{currentPageText}</VisuallyHiddenText>
+          {children}
+        </StyledCurrentLink>
+      </StyledDropdownLink>
+    ) : (
+      <StyledDropdownLink script={script} service={service} href={url}>
+        {children}
+      </StyledDropdownLink>
+    )}
+  </DropdownLi>
+);
+
+export const AmpDropdown = ({ announcedText, onClose, children }) => (
+  <DropdownWrapper>
+    <CrossButton
+      aria-label={announcedText}
+      on={onClose}
+      // eslint-disable-next-line react/jsx-boolean-value
+      aria-expanded="true"
+    >
+      {navigationIcons.cross}
+    </CrossButton>
+    <DropdownUl role="list">{children}</DropdownUl>
+  </DropdownWrapper>
+);
+
+export const CanonicalDropdown = ({ announcedText, onClose, children }) => (
+  <DropdownWrapper>
+    <CrossButton
+      aria-label={announcedText}
+      onClick={onClose}
+      // eslint-disable-next-line react/jsx-boolean-value
+      aria-expanded="true"
+    >
+      {navigationIcons.cross}
+    </CrossButton>
+    <DropdownUl role="list">{children}</DropdownUl>
+  </DropdownWrapper>
+);
+
+const HamburgerButton = styled.button`
+  width: ${MENU_ICON_SIDE_LENGTH};
+  height: ${MENU_ICON_SIDE_LENGTH};
+  background-color: ${C_POSTBOX};
+  padding: 0;
+  margin: 0;
+  border: 0;
+
+  &:focus {
+    box-shadow: inset 0 0 0 ${GEL_SPACING_HLF} ${C_WHITE};
+  }
+
+  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
+    display: none;
+  }
+`;
+
+export const AmpHamburgerMenu = ({ announcedText, onOpen }) => (
+  <HamburgerButton aria-label={announcedText} on={onOpen} aria-expanded="false">
+    {navigationIcons.hamburger}
+  </HamburgerButton>
+);
+
+export const CanonicalHamburgerMenu = ({ announcedText, onOpen }) => (
+  <HamburgerButton
+    aria-label={announcedText}
+    onClick={onOpen}
+    aria-expanded="false"
+  >
+    {navigationIcons.hamburger}
+  </HamburgerButton>
+);
+
+DropdownNavigationLi.propTypes = {
+  children: string.isRequired,
+  url: string.isRequired,
+  script: shape(scriptPropType).isRequired,
+  service: string.isRequired,
+  active: bool,
+  currentPageText: string,
+  dir: oneOf(['ltr', 'rtl']),
+};
+
+DropdownNavigationLi.defaultProps = {
+  active: false,
+  currentPageText: null,
+  dir: 'ltr',
+};
+
+AmpDropdown.propTypes = {
+  announcedText: string.isRequired,
+  children: node.isRequired,
+  onClose: func.isRequired,
+};
+
+CanonicalDropdown.propTypes = {
+  announcedText: string.isRequired,
+  children: node.isRequired,
+  onClose: func.isRequired,
+};
+
+AmpHamburgerMenu.propTypes = {
+  announcedText: string.isRequired,
+  onOpen: func.isRequired,
```

Comment:
The amp version will be a string rather than a function (see https://github.com/bbc/psammead/pull/2302/files#diff-649fabcf7d0f0c3dc0bfa5b342f106b2R431)
```suggestion
  onClose: string.isRequired,
```